remotes/docker/resolver_test.go

```diff
@@ -192,6 +192,86 @@ func TestBadTokenResolver(t *testing.T) {
 	}
 }
 
+func TestHostFailureFallbackResolver(t *testing.T) {
+	sf := func(h http.Handler) (string, ResolverOptions, func()) {
+		s := httptest.NewServer(h)
+		base := s.URL[7:] // strip "http://"
+
+		options := ResolverOptions{}
+		createHost := func(host string) RegistryHost {
+			return RegistryHost{
+				Client:       &http.Client{},
+				Host:         host,
+				Scheme:       "http",
+				Path:         "/v2",
+				Capabilities: HostCapabilityPull | HostCapabilityResolve | HostCapabilityPush,
+			}
+		}
+
+		// Override hosts with two hosts
+		options.Hosts = func(host string) ([]RegistryHost, error) {
+			return []RegistryHost{
+				createHost("127.0.0.1:1234"), // This host is NOT running
```

Comment:
The `127.0.0.1:1234` might be used for registry in the host.

I would like to use `httptest. NewUnstartedServer(nil)` to get server which start listener. The listener will randomly pick one port in host. Since `NewUnstartedServer` doesn't start server, we can make sure that the `127.0.0.1:random-port` doesn't serve for registry. WDYT?

Reply:
SGTM. I've updated the test to use `NewUnstartedServer` to generate a random port.